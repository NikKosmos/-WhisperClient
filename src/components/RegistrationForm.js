import React from 'react'
import PropTypes from 'prop-types'

class RegistrationForm extends React.Component {

    state = {
        email: '',
        name: '',
        password: '',
    }

    onSubmit = (eventArg) => {
        eventArg.preventDefault()

        const { email, name, password } = this.state

        const user = {
            email,
            name,
            password,
        }

        this.props.onSubmit(user)

        this.setState({
            email: '',
            name: '',
            password: '',
        })
    }

    updateUserEmailValue = (eventArg) => {
        this.setState({
            email: eventArg.target.value
        })
    }

    updateUserNameValue = (eventArg) => {
        this.setState({
            name: eventArg.target.value
        })
    }

    updateUserPasswordValue = (eventArg) => {
        this.setState({
            password: eventArg.target.value
        })
    }

    render() {
        return (
            <form onSubmit = { this.onSubmit } className = "regAuthForm">
                <h3>New user registration</h3>
                <label>User email</label>
                <input
                    name = "userEmail"
                    placeholder = 'User email'
                    type = 'text'
                    value = { this.state.email }
                    onChange = { this.updateUserEmailValue }
                /><br />
                <label>User name</label>
                <input
                    name = 'userName'
                    placeholder = 'User name'
                    type = 'text'
                    value = { this.state.name }
                    onChange = { this.updateUserNameValue }
                /><br />
                <label>User password</label>
                <input
                    name = 'userPassword'
                    placeholder = 'User password'
                    type = 'text'
                    value = { this.state.password }
                    onChange = { this.updateUserPasswordValue }
                /><br />
                <button type = 'submit'>
                    Send new user registration data
                </button>
            </form>
        )
    }
}

RegistrationForm.propTypes = {
    onSubmit: PropTypes.func.isRequired
}

export { RegistrationForm }