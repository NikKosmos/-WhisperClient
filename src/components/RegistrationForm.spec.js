import React from 'react'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json';

import { RegistrationForm } from './RegistrationForm'

describe('RegistrationForm', () => {
    const mockOnSubmit = jest.fn()

    it ('should render correctly', () => {
        const testedComponent = shallow(<RegistrationForm 
            onSubmit = { mockOnSubmit }
        />)

        expect(shallowToJson(testedComponent)).toMatchSnapshot();
      })

})

describe('when typing email', () => {
    const mockOnSubmit = jest.fn()
    const newTestUserEmail = 'capitan-nemo@google.com'

    const myComponent = shallow(<RegistrationForm
        onSubmit = { mockOnSubmit }
    />)

    beforeEach(() => {
        myComponent.find('[name="userEmail"]').simulate('change', {
            target: {
                value : newTestUserEmail,
            }
        })
    })

    it('updates inputValue field in state', () => {
        expect(myComponent.state().email).toEqual(newTestUserEmail);
    })

})

describe('when typing name', () => {
    const mockOnSubmit = jest.fn()
    const newTestUserName = 'Capitan Nemo'

    const myComponent = shallow(<RegistrationForm
        onSubmit = { mockOnSubmit }
    />)

    beforeEach(() => {
        myComponent.find('[name="userName"]').simulate('change', {
            target: {
                value : newTestUserName,
            }
        })
    })

    it('updates inputValue field in state', () => {
        expect(myComponent.state().name).toEqual(newTestUserName);
    })

})

describe('when typing name', () => {
    const mockOnSubmit = jest.fn()
    const newTestUserPassword = 'root'

    const myComponent = shallow(<RegistrationForm
        onSubmit = { mockOnSubmit }
    />)

    beforeEach(() => {
        myComponent.find('[name="userPassword"]').simulate('change', {
            target: {
                value : newTestUserPassword,
            }
        })
    })

    it('updates inputValue field in state', () => {
        expect(myComponent.state().password).toEqual(newTestUserPassword);
    })

})

describe('when push the button', () => {
    const mockOnSubmit = jest.fn()

    const myComponent = shallow(<RegistrationForm
        onSubmit = { mockOnSubmit }
    />)

    beforeEach(() => {
        myComponent.find('form').simulate('submit', {
            preventDefault: () => { },
        })
    })

    it('calls the props.onSubmit', () => {
        expect(mockOnSubmit).toHaveBeenCalledTimes(1);
    })

})