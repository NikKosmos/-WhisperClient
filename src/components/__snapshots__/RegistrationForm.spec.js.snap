// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h1>
    New user registration
  </h1>
  <label>
    User email
  </label>
  <input
    name="userEmail"
    onChange={[Function]}
    placeholder="User email"
    type="text"
    value=""
  />
  <br />
  <label>
    User name
  </label>
  <input
    name="userName"
    onChange={[Function]}
    placeholder="User name"
    type="text"
    value=""
  />
  <br />
  <label>
    User password
  </label>
  <input
    name="userPassword"
    onChange={[Function]}
    placeholder="User password"
    type="text"
    value=""
  />
  <br />
  <button
    type="submit"
  >
    Send new user registration data
  </button>
</form>
`;
